quarkchain/cluster/shard_state.py

```diff
@@ -1995,4 +2015,23 @@ def get_total_balance(
         limit: int,
         starter: Optional[bytes] = None,
     ) -> Tuple[int, bytes]:
-        return 42, b""
+        """
+        starter should be exclusive
+        """
+        evm_state = self._get_evm_state_from_hash(block_hash)
+
+        trie = evm_state.trie.trie
+        key = trie.next(bytes(32) if not starter else starter)
+        total = 0
+        ret = None
+        while limit > 0 and key is not None:
+            addr = evm_state.trie.db.get(key)
+            balance = evm_state.get_balance(addr, token_id, should_cache=False)
+            print(key.hex(), balance)
+            total += balance
+            ret = key
+            key = trie.next(key)
+            limit -= 1
+
+        # TODO: to decide whether to return the address form which is evm_state.trie.db.get(key)
+        return total, ret if key else bytes(32)
```

Comment:
to avoid changing our RPC interface (which passes an address as the `next/start` key), I would recommend returning the address